### ISSUE TITLE
Profiling feedback for devtools

### ISSUE THREAD

**Issue by kenzieschmoll** [flutter/devtools#1000]
feedback from @escamoteur:

> 1. Why don't I see my Dart code  in the performance or Timeline View?
> 2. Its would be great if the performance view would if the functions as waiting async and how long? (not sure if possible
> 3. To make the profileing really usefull we need not only 'total time' and 'self time' of a function but also how often this function was called and how much completeTotalTime (= totalTime * numberOfCalls ) the function took
> 4. It would great to jump from Performance views directly into the source in the IDE
> 5. On the Flutter view a live mode would be great that shows which widgets are rebuild at the moment maybe if not live a record function that tells how often a widget was rebuild.

**Comment by kenzieschmoll** [comment]
1. You should be able to see your symbols if you go far enough down in the cpu samples. If you click expand all in the Call Tree view, and go deep in the tree you should be able to see methods from your code. If you look at the Bottom Up view, your symbols will be closer to the top of the tree and perhaps easier to find. You can also try increasing the profile granularity with the dropdown at the top right in the performance screen. If you do this, a popup will come up at the top of the page - I encourage you to read the linked documentation about the tradeoffs of doing this.

2. Filed an issue here: https://github.com/flutter/devtools/issues/1001

3. This sounds like a request for a method table. We have a tracking issue for that here: https://github.com/flutter/devtools/issues/841

4. That is blocked on https://github.com/dart-lang/sdk/issues/37553.

5. Good news! We have this feature in the flutter-intelliJ plugin for IntelliJ and Android Studio. You can read documentation on that here: https://flutter.dev/docs/development/tools/android-studio#show-performance-data. We have a tracking issue to add this to DevTools here: https://github.com/flutter/devtools/issues/476.

**Comment by kenzieschmoll** [comment]
@escamoteur did my response to question 1. help?

**Comment by escamoteur** [comment]
I tried it yesterday but only with a debug build and did not see any of my methods.
Maybe I need a profile build?

**Comment by kenzieschmoll** [comment]
Yes, you should definitely be using a profile build to do performance profiling 👍 I would try increasing the profile granularity and making sure you are going all the way down to the bottom of the CPU samples.

**Comment by kenzieschmoll** [comment]
Closing this issue. Please file another issue if you are still having trouble seeing your methods in the CPU profile. Thanks.
